### DISCUSSION TITLE
What does Sinatra add If I can just interface with Rack?

### DISCUSSION THREAD

**Discussion by woodie** [puma/puma#3081] [Q&A] [answered]
I'm having a lot of fun with **Ruby Cloud Functions** which use Rack and Puma. I always try to build [a lightweight interface to rack](https://github.com/woodie/caltrain-status/blob/main/app.rb) so I can put business logic outside with proper unit tests. I want to do the same thing on **Google App Engine** standard with Ruby. I can create [a similar interface to rack](https://github.com/woodie/next-caltrain-pwa/blob/master/app.rb) but from within Sinatra.

The App Engine [entrypoint](https://github.com/woodie/next-caltrain-pwa/blob/master/app.yaml#L2) is `bundle exec ruby app.rb`. Since I'm not doing much with Sinatra, I wonder about ripping out Sinatra to **just use Rack**. In my specs I'm just using `rack/test` which is [very nice](https://github.com/woodie/next-caltrain-pwa/blob/master/spec/app_spec.rb).

**Comment by dentarg** [comment]
I recommend reading the Sinatra README to understand what it does for you: https://github.com/sinatra/sinatra

**Comment by woodie** [comment]
[This gist](https://gist.github.com/markbates/4240848) provides examples using rack directly.

**Comment by nateberkopec** [comment, marked as answer]
Puma is just a Rack app server - we're completely agnostic to what Rack-compatible framework (or not!) you use. 

One of the more important things you give up when not using a framework is that [you usually have to do your own security](https://sinatrarb.com/protection/).
